src/boss/bo4/unk_45354.c

```diff
@@ -14,7 +14,13 @@ void func_8010E0B8(void) {
     g_Entities[STAGE_ENTITY_START + UNK_ENTITY_1].ext.entSlot1.unk0 = 0;
 }
 
-INCLUDE_ASM("boss/bo4/nonmatchings/unk_45354", func_us_801C5430);
+extern s16 D_us_801D4D6C[];
+
+void func_us_801C5430(s16 a0, s16 a1) {
+    if (D_us_801D4D6C[0] <= a1) {
+        D_us_801D4D6C[0] = a1;
```

Comment:
I think this might be part of `g_Dop.timers` array?
This one should match with `g_Dop.timers[14]`

To work this out I did:
* symbols.us.bobo4.txt -> g_Dop = 0x801D4A20
* `g_Dop` is a `PlayerState` of size `0x3D0` (see `game.h`)
* `801D4D6C` - `801D4A20` = 34C which is in the `timers` array of `PlayerState`